### PR TITLE
notation: init at 1.0.0-rc.7

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Janik-Haag — APPROVED** [review]
LGTM

File: pkgs/tools/security/notation/default.nix
Comment (NickCao) [inline pkgs/tools/security/notation/default.nix:16]: Please add a comment about why tests are excluded.

File: pkgs/tools/security/notation/default.nix
Comment (Janik-Haag) [inline pkgs/tools/security/notation/default.nix:17]: The comment shouldn't state what it is doing in this case disabling the tests, that's quite obvious but rather why they are disabled, like failing because the need an internet connection or stuff like that.
When I build locally with tests enabled they fail because of this:
```
main module (github.com/notaryproject/notation) does not contain package github.com/notaryproject/notation/test/e2e/internal/notation
```

**Review by Janik-Haag — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/notation/default.nix
Comment (Janik-Haag) [reply pkgs/tools/security/notation/default.nix]: maybe try ;)
> this is a Go sub-module and cannot be built directly

Comment (Janik-Haag) [reply pkgs/tools/security/notation/default.nix]: I would merge this once you have changed the comment, hope that's fine with you.

Comment (aaronjheng) [reply pkgs/tools/security/notation/default.nix]: Done. Thanks.

Comment (Janik-Haag) [reply pkgs/tools/security/notation/default.nix]: now the last thing to do is waiting for ci :D

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**
